Weird error issue with a git remote

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

On a fresh install (Linux, Debian), using the following git remote from an existing repository:
```
origin	talisker:Sources/Git/qemu (fetch)
origin	talisker:Sources/Git/qemu (push)
talisker-eth	talisker-eth:Sources/Git/qemu (fetch)
talisker-eth	talisker-eth:Sources/Git/qemu (push)
```
Prompt shows:
`  eblot   qemu : error calling url: parse "https://talisker-eth:Sources/Git/qemu": invalid port ":Sources" after host  `

Is it the dash character in the remote source name that creates this error?

### Theme

Default from installation

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

zsh

### Log output

```Shell
Version: 14.19.3

Shell: zsh (5.8)

Segments:

ConsoleTitle(true)                        -   0 ms - zsh in qemu
session(true)                             -   0 ms -   eblot 
path(true)                                -   0 ms -   qemu 
git(true)                                 -  32 ms - : error calling url: parse "https://talisker-eth:Sources/Git/qemu": invalid port ":Sources" after host
root(false)                               -   0 ms - 
exit(true)                                -   0 ms -   
node(false)                               -   0 ms - 
go(false)                                 -   0 ms - 
python(false)                             -   0 ms - 
shell(true)                               -   0 ms - in zsh 
time(true)                                -   0 ms - at 15:05:40

Run duration: 35.250377ms

Cache path: /home/eblot/.cache/oh-my-posh

Config path: 

Logs:

[DEBUG] 15:05:40.420 shell.go:Getenv:392 → NO DATA
[TRACE] 15:05:40.420 shell.go:Getenv(TERM_PROGRAM) - 15.479µs
[DEBUG] 15:05:40.420 shell.go:Getenv:392 → NO DATA
[TRACE] 15:05:40.420 shell.go:Getenv(XDG_CACHE_HOME) - 2.845µs
[TRACE] 15:05:40.420 shell_unix.go:CachePath() - 12.924µs
[DEBUG] 15:05:40.420 shell.go:Getenv:392 → NO DATA
[TRACE] 15:05:40.420 shell.go:Getenv(POSH_THEME) - 3.757µs
[DEBUG] 15:05:40.420 shell.go:resolveConfigPath:324 → No config set, fallback to default config
[TRACE] 15:05:40.420 shell.go:resolveConfigPath() - 14.086µs
[TRACE] 15:05:40.420 shell.go:Init() - 138.39µs
[TRACE] 15:05:40.420 shell.go:Flags() - 50ns
[DEBUG] 15:05:40.420 config.go:loadConfig:105 → no config file specified, using default
[TRACE] 15:05:40.420 config.go:loadConfig() - 23.113µs
[TRACE] 15:05:40.421 shell.go:Flags() - 40ns
[DEBUG] 15:05:40.421 shell.go:Getenv:392 → NO DATA
[TRACE] 15:05:40.421 shell.go:Getenv(OMP_CACHE_DISABLED) - 2.695µs
[DEBUG] 15:05:40.421 shell.go:Shell:641 → process name: zsh
[TRACE] 15:05:40.421 shell.go:Shell() - 894.059µs
[DEBUG] 15:05:40.421 shell.go:Getenv:392 → 5.8
[TRACE] 15:05:40.421 shell.go:Getenv(POSH_SHELL_VERSION) - 3.026µs
[TRACE] 15:05:40.422 shell_unix.go:Root() - 291ns
[TRACE] 15:05:40.422 shell.go:Shell() - 100ns
[TRACE] 15:05:40.422 shell.go:ErrorCode() - 51ns
[DEBUG] 15:05:40.422 shell.go:Getenv:392 → NO DATA
[TRACE] 15:05:40.422 shell.go:Getenv(WSL_DISTRO_NAME) - 2.395µs
[TRACE] 15:05:40.422 shell_unix.go:IsWsl() - 5.099µs
[TRACE] 15:05:40.422 shell.go:GOOS() - 60ns
[DEBUG] 15:05:40.422 shell.go:Pwd:420 → NO DATA
[TRACE] 15:05:40.422 shell.go:Pwd() - 13.856µs
[TRACE] 15:05:40.422 shell.go:GOOS() - 40ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 40ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 40ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 40ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 40ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 40ns
[DEBUG] 15:05:40.422 shell.go:User:545 → eblot
[TRACE] 15:05:40.422 shell.go:User() - 2.785µs
[DEBUG] 15:05:40.422 shell.go:Host:557 → keembay
[TRACE] 15:05:40.422 shell.go:Host() - 4.478µs
[TRACE] 15:05:40.422 shell.go:GOOS() - 40ns
[DEBUG] 15:05:40.422 shell_unix.go:Platform:63 → debian
[TRACE] 15:05:40.422 shell.go:TemplateCache() - 75.793µs
[TRACE] 15:05:40.422 shell.go:Flags() - 60ns
[DEBUG] 15:05:40.422 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.422 shell.go:Pwd() - 3.176µs
[TRACE] 15:05:40.422 shell.go:GOOS() - 40ns
[DEBUG] 15:05:40.422 shell.go:Getenv:392 → 192.168.25.45 52498 192.168.25.8 22
[TRACE] 15:05:40.422 shell.go:Getenv(SSH_CONNECTION) - 2.685µs
[TRACE] 15:05:40.422 shell.go:TemplateCache() - 90ns
[TRACE] 15:05:40.422 shell.go:TemplateCache() - 181ns
[TRACE] 15:05:40.422 shell.go:Shell() - 61ns
[DEBUG] 15:05:40.422 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.422 shell.go:Pwd() - 4.779µs
[TRACE] 15:05:40.422 shell.go:GOOS() - 70ns
[DEBUG] 15:05:40.422 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.422 shell.go:Pwd() - 3.547µs
[TRACE] 15:05:40.422 shell.go:Shell() - 50ns
[TRACE] 15:05:40.422 shell.go:Shell() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:PathSeparator() - 70ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:PathSeparator() - 50ns
[TRACE] 15:05:40.422 shell.go:PathSeparator() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 51ns
[TRACE] 15:05:40.422 shell.go:PathSeparator() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:PathSeparator() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:PathSeparator() - 60ns
[TRACE] 15:05:40.422 shell.go:PathSeparator() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 60ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.422 shell.go:PathSeparator() - 50ns
[TRACE] 15:05:40.422 shell.go:PathSeparator() - 50ns
[DEBUG] 15:05:40.422 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.422 shell.go:Pwd() - 3.417µs
[DEBUG] 15:05:40.422 shell.go:Getenv:392 → NO DATA
[TRACE] 15:05:40.422 shell.go:Getenv(WSL_DISTRO_NAME) - 2.895µs
[TRACE] 15:05:40.422 shell_unix.go:IsWsl() - 6.022µs
[TRACE] 15:05:40.422 shell.go:StackCount() - 91ns
[TRACE] 15:05:40.422 shell_unix.go:DirIsWritable(/home/eblot/Sources/Git/qemu) - 7.433µs
[TRACE] 15:05:40.422 shell.go:TemplateCache() - 70ns
[TRACE] 15:05:40.422 shell.go:TemplateCache() - 60ns
[TRACE] 15:05:40.422 shell.go:Shell() - 50ns
[DEBUG] 15:05:40.422 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.422 shell.go:Pwd() - 4.468µs
[TRACE] 15:05:40.423 shell.go:GOOS() - 110ns
[DEBUG] 15:05:40.423 shell.go:Getenv:392 → NO DATA
[TRACE] 15:05:40.423 shell.go:Getenv(WSL_DISTRO_NAME) - 19.206µs
[TRACE] 15:05:40.423 shell_unix.go:IsWsl() - 24.686µs
[TRACE] 15:05:40.423 shell_unix.go:IsWsl2() - 28.544µs
[TRACE] 15:05:40.423 shell.go:GOOS() - 70ns
[DEBUG] 15:05:40.423 shell.go:CommandPath:596 → /usr/bin/git
[TRACE] 15:05:40.423 shell.go:CommandPath(git) - 48.891µs
[TRACE] 15:05:40.423 shell.go:HasCommand(git) - 56.115µs
[DEBUG] 15:05:40.423 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.423 shell.go:Pwd() - 11.001µs
[TRACE] 15:05:40.423 shell.go:HasParentFilePath(.git) - 30.958µs
[TRACE] 15:05:40.423 shell.go:GOOS() - 70ns
[TRACE] 15:05:40.423 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.423 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.423 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.423 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.423 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.423 shell.go:GOOS() - 50ns
[DEBUG] 15:05:40.423 shell.go:Getenv:392 → NO DATA
[TRACE] 15:05:40.423 shell.go:Getenv(POSH_GIT_STATUS) - 6.362µs
[TRACE] 15:05:40.423 shell.go:GOOS() - 50ns
[DEBUG] 15:05:40.454 shell.go:RunCommand:575 ↓
    # branch.oid d46c7851d8cd78f3ca813e288df9febdc6815920
    # branch.head dev/opentitan
    # branch.upstream talisker-eth/dev/opentitan
    # branch.ab +39 -39
    1 .M N... 100644 100644 100644 0f0c81031aed4ea003781240746e713edac0047e 0f0c81031aed4ea003781240746e713edac0047e hw/opentitan/ot_pinmux.c
    ? log.txt
[TRACE] 15:05:40.454 shell.go:RunCommand(git -C /home/eblot/Sources/Git/qemu --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --branch --porcelain=2) - 30.538902ms
[DEBUG] 15:05:40.454 shell.go:HasFolder:484 → false
[TRACE] 15:05:40.454 shell.go:HasFolder(/home/eblot/Sources/Git/qemu/.git/rebase-merge) - 5.871µs
[DEBUG] 15:05:40.454 shell.go:HasFolder:484 → false
[TRACE] 15:05:40.454 shell.go:HasFolder(/home/eblot/Sources/Git/qemu/.git/rebase-apply) - 3.326µs
[DEBUG] 15:05:40.454 shell.go:HasFilesInDir:455 → false
[TRACE] 15:05:40.454 shell.go:HasFilesInDir(MERGE_MSG) - 7.794µs
[DEBUG] 15:05:40.454 shell.go:HasFilesInDir:455 → false
[TRACE] 15:05:40.454 shell.go:HasFilesInDir(CHERRY_PICK_HEAD) - 3.878µs
[DEBUG] 15:05:40.454 shell.go:HasFilesInDir:455 → false
[TRACE] 15:05:40.454 shell.go:HasFilesInDir(REVERT_HEAD) - 3.516µs
[DEBUG] 15:05:40.454 shell.go:HasFilesInDir:455 → false
[TRACE] 15:05:40.454 shell.go:HasFilesInDir(sequencer/todo) - 5.109µs
[TRACE] 15:05:40.454 shell.go:TemplateCache() - 111ns
[TRACE] 15:05:40.454 shell.go:TemplateCache() - 70ns
[ERROR] 15:05:40.454 text.go:Render:61 → template:  {{ if .Data.UpstreamURL }}{{ url .Data.UpstreamIcon .Data.UpstreamURL }} {{ end }}{{ .Data.HEAD }}{{if .Data.BranchStatus }} {{ .Data.BranchStatus }}{{ end }}{{ if .Data.Working.Changed }}  {{ .Data.Working.String }}{{ end }}{{ if .Data.Staging.Changed }}  {{ .Data.Staging.String }}{{ end }} :1:30: executing " {{ if .Data.UpstreamURL }}{{ url .Data.UpstreamIcon .Data.UpstreamURL }} {{ end }}{{ .Data.HEAD }}{{if .Data.BranchStatus }} {{ .Data.BranchStatus }}{{ end }}{{ if .Data.Working.Changed }} \uf044 {{ .Data.Working.String }}{{ end }}{{ if .Data.Staging.Changed }} \uf046 {{ .Data.Staging.String }}{{ end }} " at <url .Data.UpstreamIcon .Data.UpstreamURL>: error calling url: parse "https://talisker-eth:Sources/Git/qemu": invalid port ":Sources" after host
[TRACE] 15:05:40.454 shell.go:Shell() - 100ns
[TRACE] 15:05:40.454 shell.go:TemplateCache() - 71ns
[TRACE] 15:05:40.455 shell.go:TemplateCache() - 90ns
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 3.256µs
[TRACE] 15:05:40.455 shell.go:GOOS() - 50ns
[TRACE] 15:05:40.455 shell_unix.go:Root() - 391ns
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 2.374µs
[TRACE] 15:05:40.455 shell.go:GOOS() - 30ns
[TRACE] 15:05:40.455 shell.go:ErrorCode() - 40ns
[TRACE] 15:05:40.455 shell.go:TemplateCache() - 60ns
[TRACE] 15:05:40.455 shell.go:TemplateCache() - 40ns
[TRACE] 15:05:40.455 shell.go:Shell() - 50ns
[TRACE] 15:05:40.455 shell.go:TemplateCache() - 40ns
[TRACE] 15:05:40.455 shell.go:Flags() - 50ns
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 5.501µs
[TRACE] 15:05:40.455 shell.go:GOOS() - 30ns
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 2.444µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 3.647µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(*.js) - 110.537µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 2.204µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(*.ts) - 86.732µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 2.134µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(package.json) - 9.207µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.944µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(.nvmrc) - 9.578µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.974µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(pnpm-workspace.yaml) - 7.404µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 2.004µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(.pnpmfile.cjs) - 10.329µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 2.063µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(.npmrc) - 7.353µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.844µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(.vue) - 5.961µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.563µs
[TRACE] 15:05:40.455 shell.go:GOOS() - 30ns
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.583µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.452µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(*.go) - 59.581µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.503µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(go.mod) - 6.883µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 2.064µs
[TRACE] 15:05:40.455 shell.go:GOOS() - 30ns
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.553µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.423µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(*.py) - 55.985µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 2.956µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(*.ipynb) - 70.663µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 2.034µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(pyproject.toml) - 8.637µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.482µs
[DEBUG] 15:05:40.455 shell.go:HasFiles:441 → false
[TRACE] 15:05:40.455 shell.go:HasFiles(venv.bak) - 5.63µs
[DEBUG] 15:05:40.455 shell.go:HasFolder:484 → false
[TRACE] 15:05:40.455 shell.go:HasFolder(.venv) - 2.204µs
[DEBUG] 15:05:40.455 shell.go:HasFolder:484 → false
[TRACE] 15:05:40.455 shell.go:HasFolder(venv) - 1.974µs
[DEBUG] 15:05:40.455 shell.go:HasFolder:484 → false
[TRACE] 15:05:40.455 shell.go:HasFolder(virtualenv) - 3.276µs
[DEBUG] 15:05:40.455 shell.go:HasFolder:484 → false
[TRACE] 15:05:40.455 shell.go:HasFolder(env) - 1.884µs
[DEBUG] 15:05:40.455 shell.go:HasFolder:484 → false
[TRACE] 15:05:40.455 shell.go:HasFolder(venv-win) - 1.873µs
[DEBUG] 15:05:40.455 shell.go:HasFolder:484 → false
[TRACE] 15:05:40.455 shell.go:HasFolder(pyenv-win) - 1.854µs
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.684µs
[TRACE] 15:05:40.455 shell.go:GOOS() - 20ns
[TRACE] 15:05:40.455 shell.go:Shell() - 41ns
[TRACE] 15:05:40.455 shell.go:Flags() - 20ns
[TRACE] 15:05:40.455 shell.go:TemplateCache() - 50ns
[TRACE] 15:05:40.455 shell.go:TemplateCache() - 30ns
[TRACE] 15:05:40.455 shell.go:Shell() - 30ns
[DEBUG] 15:05:40.455 shell.go:Pwd:400 → /home/eblot/Sources/Git/qemu
[TRACE] 15:05:40.455 shell.go:Pwd() - 1.773µs
[TRACE] 15:05:40.455 shell.go:GOOS() - 20ns
[TRACE] 15:05:40.455 shell.go:TemplateCache() - 31ns
[TRACE] 15:05:40.456 shell.go:TemplateCache() - 30ns
[TRACE] 15:05:40.456 shell.go:Shell() - 50ns
[DEBUG] 15:05:40.456 shell.go:Getenv:392 → NO DATA
[TRACE] 15:05:40.456 shell.go:Getenv(XDG_CACHE_HOME) - 1.874µs
[TRACE] 15:05:40.456 shell_unix.go:CachePath() - 7.394µs
[TRACE] 15:05:40.456 shell.go:Flags() - 30ns
```

--- a/src/segments/git.go
+++ b/src/segments/git.go
@@ -388,26 +388,42 @@ func (g *Git) setBranchStatus() {
 	g.BranchStatus = getBranchStatus()
 }
 
-func (g *Git) getUpstreamIcon() string {
-	cleanSSHURL := func(url string) string {
-		if strings.HasPrefix(url, "http") {
-			return url
-		}
-		url = strings.TrimPrefix(url, "ssh://")
-		url = strings.TrimPrefix(url, "git://")
-		if i := strings.Index(url, "@"); i >= 0 {
-			url = url[i+1:]
-			url = strings.Replace(url, ":", "/", 1)
-		}
-		url = strings.TrimSuffix(url, ".git")
-		return fmt.Sprintf("https://%s", url)
+func (g *Git) cleanUpstreamURL(url string) string {
+	if strings.HasPrefix(url, "http") {
+		return url
 	}
+	// /path/to/repo.git/
+	match := regex.FindNamedRegexMatch(`^(?P<URL>[a-z0-9./]+)$`, url)
+	if len(match) != 0 {
+		url := strings.Trim(match["URL"], "/")
+		url = strings.TrimSuffix(url, ".git")
+		return fmt.Sprintf("https://%s", strings.TrimPrefix(url, "/"))
+	}
+	// ssh://user@host.xz:1234/path/to/repo.git/
+	match = regex.FindNamedRegexMatch(`(ssh|ftp|git|rsync)://(.*@)?(?P<URL>[a-z0-9.]+)(:[0-9]{4})?/(?P<PATH>.*).git`, url)
+	if len(match) == 0 {
+		// host.xz:/path/to/repo.git/
+		match = regex.FindNamedRegexMatch(`^(?P<URL>[a-z0-9./]+):(?P<PATH>[a-z0-9./]+)$`, url)
+	}
+	if len(match) != 0 {
+		path := strings.Trim(match["PATH"], "/")
+		path = strings.TrimSuffix(path, ".git")
+		return fmt.Sprintf("https://%s/%s", match["URL"], path)
+	}
+	// user@host.xz:/path/to/repo.git
+	match = regex.FindNamedRegexMatch(`.*@(?P<URL>.*):(?P<PATH>.*).git`, url)
+	if len(match) == 0 {
+		return ""
+	}
+	return fmt.Sprintf("https://%s/%s", match["URL"], match["PATH"])
+}
 
+func (g *Git) getUpstreamIcon() string {
 	g.RawUpstreamURL = g.getRemoteURL()
 	if len(g.RawUpstreamURL) == 0 {
 		return ""
 	}
-	g.UpstreamURL = cleanSSHURL(g.RawUpstreamURL)
+	g.UpstreamURL = g.cleanUpstreamURL(g.RawUpstreamURL)
 
 	// allow overrides first
 	custom := g.props.GetKeyValueMap(UpstreamIcons, map[string]string{})

--- a/src/segments/git_test.go
+++ b/src/segments/git_test.go
@@ -589,6 +589,31 @@ func TestGetStashContextZeroEntries(t *testing.T) {
 	}
 }
 
+func TestGitCleanSSHURL(t *testing.T) {
+	cases := []struct {
+		Case     string
+		Expected string
+		Upstream string
+	}{
+		{Case: "regular URL", Expected: "https://src.example.com/user/repo", Upstream: "/src.example.com/user/repo.git"},
+		{Case: "domain:path", Expected: "https://host.xz/path/to/repo", Upstream: "host.xz:/path/to/repo.git/"},
+		{Case: "ssh with port", Expected: "https://host.xz/path/to/repo", Upstream: "ssh://user@host.xz:1234/path/to/repo.git"},
+		{Case: "ssh with port, trailing slash", Expected: "https://host.xz/path/to/repo", Upstream: "ssh://user@host.xz:1234/path/to/repo.git/"},
+		{Case: "ssh without port", Expected: "https://host.xz/path/to/repo", Upstream: "ssh://user@host.xz/path/to/repo.git/"},
+		{Case: "ssh port, no user", Expected: "https://host.xz/path/to/repo", Upstream: "ssh://host.xz:1234/path/to/repo.git"},
+		{Case: "ssh no port, no user", Expected: "https://host.xz/path/to/repo", Upstream: "ssh://host.xz/path/to/repo.git"},
+		{Case: "rsync no port, no user", Expected: "https://host.xz/path/to/repo", Upstream: "rsync://host.xz/path/to/repo.git/"},
+		{Case: "git no port, no user", Expected: "https://host.xz/path/to/repo", Upstream: "git://host.xz/path/to/repo.git"},
+		{Case: "gitea no port, no user", Expected: "https://src.example.com/user/repo", Upstream: "_gitea@src.example.com:user/repo.git"},
+		{Case: "unsupported", Upstream: "\\test\\repo.git"},
+	}
+	for _, tc := range cases {
+		g := &Git{}
+		upstreamIcon := g.cleanUpstreamURL(tc.Upstream)
+		assert.Equal(t, tc.Expected, upstreamIcon, tc.Case)
+	}
+}
+
 func TestGitUpstream(t *testing.T) {
 	cases := []struct {
 		Case     string

